Wangscape/noise/Reseedable.h

```diff
@@ -58,8 +58,8 @@ struct Reseedable final
 template<typename InputIt>
 inline Reseedable Reseedable::terrace(InputIt first, InputIt last, bool inverted)
 {
-    static_assert(std::is_same<std::iterator_traits<InputIt>::iterator_category::input_iterator_tag,
-                  std::input_iterator_tag>::value,
+    static_assert(std::is_same<typename std::iterator_traits<InputIt>::iterator_category,
```

Comment:
For me this prevents `TestReseedable::TestReseedableTerraceManual` from compiling, because `std::vector::const_iterator` is not an `InputIterator`. This should be implemented with `std::is_base_of` or `std::convertible_to`.